ls1intum/Artemis
# `Programming exercises`: Export and import embedded files in problem statement

### Checklist
#### General


#### Server

#### Changes affecting Programming Exercises

### Motivation and Context


Closes  #6535

### Description
Until now files embedded into the problem statement are not part of the export. This is not a problem as long as the export and import happens on the same instance. Once one imports a programming exercise on a different instance, the embedded files and hence parts of the problem statements are missing.
This PR adds the embedded files to the export and considers them when importing a programming exercise from a file

### Steps for Testing

Prerequisites:
- 1 Instructor

1. Log in to Artemis
2. Navigate to Course Administration
3.  Create a programming exercise with a  problem statement that contains one or more embedded files (e.g. jpg files)
4. Export the exercise (Download exercise button)
5. Use another test server to import the exercise from file (make sure to deploy this branch first)
6. After the import the files should be shown in the problem statement (during the import they cannot be shown as they do not yet exist on the server)

#### Exam Mode Testing



Prerequisites:
- 1 Instructor
-  Another instance as the one used for the export (make sure to deploy this branch there)

1. Log in to Artemis
2. Import the programming exercise from file you previously exported into an exercise group
3. After the import the files should be shown in the problem statement (during the import they cannot be shown as they do not yet exist on the server)

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests
#### Exam Mode Test

### Test Coverage



| Class/File | Line Coverage | Confirmation (assert/expect) | Coment |
|------------|--------------:|-----------------------------:|----------------:|
| ProgrammingExerciseImportFromFileService.java | 96% | ✅ |
| ProgrammingExerciseExportService | 86% | ✅ | my changes are fully covered |
|ZipFileService.java| 96%| ✅ |
|programming-exercise-problem.component.ts| 100%| ✅ |
|programming-exercise-update-wizard-problem.component.ts|100% |✅|
